libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -564,6 +566,26 @@ MQTTStatus_t MQTT_DeserializeAck( const MQTTPacketInfo_t * pIncomingPacket,
                                   uint16_t * pPacketId,
                                   bool * pSessionPresent );
 
+/**
+ * @brief Parses the payload of an MQTT SUBACK packet that contains the status
+ * code of subscription request made for topic filters in the equivalent
+ * subscribe packet.
+ *
+ * This function can be used to iterate over the status code
+ *
+ * @param[in] pSubackPacket The SUBACK packet whose payload is to be parsed.
+ * @param[out] pPayloadStart This is populated with the starting address
+ * of the payload (or return codes for topic filters) in the SUBACK packet.
+ * @param[out] pPayloadSize This is populated with the size of the payload
+ * in the SUBACK packet. It represents the number of topic filters whose
+ * SUBACK status is present in the packet.
+ *
+ * @return  #MQTTBadParameter, or #MQTTSuccess.
+ */
+MQTTStatus_t MQTT_GetSubAckPayload( MQTTPacketInfo_t * pSubackPacket,
```

Comment:
Should this be in `mqtt_lightweight.h`? Although it doesn't use an `MQTTContext_t`, I thought the purpose of creating a new API function was so the application wouldn't have to mix lightweight and managed API. If that's the case, I'd expect this to be in `mqtt.h`

Reply:
Okay, (imo) the function provides a packet-layer parsing utility which logically belongs to the lightweight API. 
However, based on suggestion from @abhidixi11 as well, have moved it to the `mqtt.h` layer